docs/modules/ROOT/pages/customizing-entities/entity-validation.adoc

```diff
@@ -0,0 +1,36 @@
+= Customizing entity validation
+
+By default annotation validation is performed on all entities.
+Customizing validation can be done by simply specifying a `Validator` bean that supports the specific entity type.
+You can use the `EntityValidatorSupport` as a base class to extend the default annotation based entity validation.
+
+.Custom entity validator
+[source,java,indent=0]
+[subs="verbatim,quotes,attributes"]
+----
+@Component
+public class SeatValidator extends EntityValidatorSupport<Seat> {
+
+    @Override
+    protected void preValidation(Seat entity, Errors errors, Object... validationHints) { <1>
+        // Trigger validation before the initial enitty validation happens
```

Comment:
I would suggest using either comments in the code snippet or callouts, but not both together